nixos/modules/services/hardware/dockd.nix

```diff
@@ -29,65 +31,131 @@ in
-      default = "/etc/dockd/undock.hook";
+      default = "";
       description = ''
-        Path to the executable to trigger when the ThinkPad is removed
-        from the dock.
-        Ignored if the "undockHook" option is set.
+        Path to the executable to run when the ThinkPad is removed from the dock.
+
+        Example: "/etc/dockd/undock.hook"
+
+        If set, make sure the file exists and is executable by the user of the session.
+
+        This option is mutually exclusive to the "undockHook" option.
       '';
     };
 
     undockHook = mkOption {
-      type = types.str;
-      default = "";
+      type = with types; nullOr str;
+      default = null;
       description = ''
         Script to be executed when the ThinkPad is removed from the dock.
-        If defined, the "undockHookPath" option will be ignored.
+        If is set to null (the default) then create dummy script to
+        fulfill the expectations of dockd.
+
+        Example:  "echo undocked"
+
+        This option is mutually exclusive to the "undockHookPath" option.
       '';
     };
 
   };
   ##################################################
   config =
-  let
-    # override the paths on which the dockd executable
-    # should lookup for the configuration and hook scripts
-    dockd =
-     pkgs.dockd.override {
-       inherit (cfg) dockedConfPath undockedConfPath;
-       dockHookPath = (if cfg.dockHook != ""
-                       then pkgs.writeScript "dock.hook"
-                       else cfg.dockHookPath);
-       undockHookPath = (if cfg.undockHook != ""
-                         then pkgs.writeScript "undock.hook" cfg.undockHook
-                         else cfg.undockHookPath);
-     };
-  in mkIf cfg.enable {
+    let
+       # override the paths on which the dockd executable
+       # should lookup for the configuration and hook scripts
+       dockd =
+        # configure a default script content if the dockHook or undockHook hook is
+        # set to null, specified that the user is not interested on doing nothing
+        # with the hooks
+        let
+          dockHook = if cfg.dockHook == null then "exit 0" else cfg.dockHook;
+          undockHook = if cfg.undockHook == null then "exit 0" else cfg.undockHook;
+          inherit(pkgs) writeScript;
+        in
+        pkgs.dockd.override {
+          inherit (cfg) dockedConfPath undockedConfPath;
+          dockHookPath = (if dockHook  != ""
+                          then writeScript "dock.hook" dockHook
+                          else cfg.dockHookPath);
+          undockHookPath = (if undockHook  != ""
+                            then writeScript "undock.hook" undockHook
+                            else cfg.undockHookPath);
+        };
+    in
+    mkIf cfg.enable {
+    assertions = [
+      # verify that only one of (dockHook | dockHookPath) is defined
+      {
+        assertion =
+         if ((cfg.dockHook != null && cfg.dockHook != "") && (cfg.dockHookPath != ""))
```

Comment:
This can be simplified to `assertion = dockHook == "" || dockHookPath == "";` (same for undock)
(removing null as suggested and simplify)

Reply:
This was the original approach, but changed because of the idea of `null` intending "I don't care" and a string that you want to evaluate this as a script. I can make as it was before.